internal/cloud/keyvault.go

```diff
@@ -55,134 +53,103 @@ func getAzureClientAuthMethod() clientAuthMethod {
 	return mode
 }
 
-// NewAzureClient returns a new Azure Key Vault client
-// By default, the authorizer will be from Managed Identity
-// If user sets AKV_AUTH_METHOD to AKV_AUTH_FROM_CLI, it will use Azure CLI instead
-func NewAzureClient() (*keyvault.BaseClient, error) {
-	var (
-		authorizer autorest.Authorizer
-		err        error
-	)
+// AzureCredential returns an Azure TokenCredential
+func AzureCredential() (credential azcore.TokenCredential, err error) {
 	authMethod := getAzureClientAuthMethod()
 	switch authMethod {
 	case authorizerFromMI:
-		authorizer, err = auth.NewAuthorizerFromEnvironment()
+		credential, err = azidentity.NewManagedIdentityCredential(nil)
 		if err != nil {
-			return nil, fmt.Errorf("%s, origin error: %w", errMsgAuthorizerFromMI, err)
+			err = fmt.Errorf("%s. Original error: %w", errMsgAuthorizerFromMI, err)
 		}
 	case authorizerFromCLI:
-		authorizer, err = auth.NewAuthorizerFromCLI()
+		credential, err = azidentity.NewAzureCLICredential(nil)
 		if err != nil {
-			return nil, fmt.Errorf("%s, origin error: %w", errMsgAuthorizerFromCLI, err)
+			err = fmt.Errorf("%s. Original error: %w", errMsgAuthorizerFromCLI, err)
 		}
 	default:
-		return nil, errors.New(errMsgUnknownAuthorizer)
+		err = errors.New(errMsgUnknownAuthorizer)
 	}
-	client := keyvault.New()
-	client.Authorizer = authorizer
-	return &client, nil
+	return
 }
 
-// Key represents a remote key in the Azure Key Vault.
-type Key struct {
-	Client *keyvault.BaseClient
+// KeyVault represents a Azure KeyVault Vault.
+type KeyVault struct {
+	keyClient    *azkeys.Client
+	certClient   *azcertificates.Client
+	secretClient *azsecrets.Client
 
-	vaultBaseURL string
-	name         string
-	version      string
+	name    string
+	version string
 }
 
-// NewKeyFromID create a remote key referenced by a key identifier.
-func NewKeyFromID(client *keyvault.BaseClient, keyID string) (*Key, error) {
-	keyURL, err := url.Parse(keyID)
+// NewKeyVault function creates a new instance of KeyVault struct
+func NewKeyVault(vaultName, dnsSuffix, keyName, version string) (*KeyVault, error) {
```

Comment:
We can keep the original constructor form
```go
func NewCertificateFromID(id string) (*Certificate, error) {
```
We expect `id` to be a certificate ID. For backward compatibility, key ID is also accepted. Meanwhile, we need to review all the related docs. /cc @FeynmanZhou @yizha1